src/lib.rs

```diff
@@ -1,14 +1,98 @@
+use rand::Rng;
+
 mod totp;
 mod hotp;
 
+pub struct SecretKey {
+    ascii: Option<String>,
+    hex: Option<String>,
+    base32: Option<String>,
+    otpauth_url: Option<String>
+}
+
 pub fn digest() {}
-pub fn generate_secret(){}
+
+pub fn generate_secret(
+    length: Option<u32>,
+    symbols: Option<bool>,
+    otpauth_url: Option<bool>,
+    name: Option<&str>,
+    issuer: Option<&str>,
+) -> SecretKey {
+    SecretKey { ascii: None, hex: None, base32: None, otpauth_url: None } // TODO (kevinburchfield) - fix. Only did this so it would compile
```

Comment:
good for now, but after looking at the function, i think having this generate an otpauth url is a little odd. Id rather split that functionality up into separate functions. What do you think?

Reply:
Yeah, that makes sense. speakeasy has the generateSecret function just calling another function if that option is passed so it's no problem to pull it out.

Reply:
Pulled it out and stubbed the otpauth function